### PR TITLE
Add Api to trigger manual refresh for dataset

### PR DESCRIPTION
**Pull request by sgrebnov**
Adds `v1/datasets/:name/refresh` Api endpoint to trigger manual dataset refresh.

Refresh logic is not implemented yet; `501 Not Implemented` is currently returned.

```
curl -i -X POST 127.0.0.1:3000/v1/datasets/taxi_trips/refresh
```

### CODE REVIEW

File: crates/runtime/src/http/v1.rs [hunk 1]
```diff
@@ -387,6 +388,53 @@ pub(crate) mod datasets {
...
+            return (status::StatusCode::INTERNAL_SERVER_ERROR).into_response();
+        };
+
+        let Some(dataset) = readable_app
+            .datasets
+            .iter()
+            .find(|d| d.name == dataset_name)
+        else {
+            return (
+                status::StatusCode::BAD_REQUEST,
```
Comment (digadeesh) [inline crates/runtime/src/http/v1.rs]: Probably 404 is more appropriate here.

Comment (sgrebnov) [reply crates/runtime/src/http/v1.rs]: Yeah, good point @digadeesh - thank you. Double checked and GitHub API also returns a 404 when a specific repository, user, or other resource specified in the URL path does not exist.

Comment (sgrebnov) [reply crates/runtime/src/http/v1.rs]: Updated 👍

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/src/http/v1.rs [hunk 2]
```diff
@@ -387,6 +388,53 @@ pub(crate) mod datasets {
...
+    ) -> Response {
+        let app_lock = app.read().await;
+        let Some(readable_app) = &*app_lock else {
+            return (status::StatusCode::INTERNAL_SERVER_ERROR).into_response();
+        };
+
+        let Some(dataset) = readable_app
+            .datasets
+            .iter()
+            .find(|d| d.name == dataset_name)
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs]: This should probably be a case-insensitive compare.

File: crates/runtime/src/http/v1.rs [hunk 3]
```diff
@@ -387,6 +388,53 @@ pub(crate) mod datasets {
...
+        };
+
+        let acceleration_enabled = dataset.acceleration.as_ref().is_some_and(|f| f.enabled);
+
+        if !acceleration_enabled {
+            return (
+                status::StatusCode::BAD_REQUEST,
+                Json(DatasetRefreshResponse {
+                    message: format!(
+                        "Dataset {dataset_name} does not have local acceleration enabled"
```
Comment (lukekim) [inline crates/runtime/src/http/v1.rs]:
```suggestion
                        "Dataset {dataset_name} does not have acceleration enabled"
```

File: crates/runtime/src/http/v1.rs (hunk 2 above)
Comment (sgrebnov) [reply crates/runtime/src/http/v1.rs]: Updated - thank you!

File: crates/runtime/src/http/v1.rs (hunk 3 above)
Comment (sgrebnov) [reply crates/runtime/src/http/v1.rs]: Updated - thank you!

**Review by digadeesh — DISMISSED** [review]

**Review by digadeesh — APPROVED** [review]

**Pull request merged**